homeassistant/components/libre_hardware_monitor/coordinator.py

```diff
@@ -65,6 +65,24 @@ def __init__(
             )
         )
 
+        # update devices created before 2025.10.2 without unique device id
+        if legacy_device_ids := [
+            device_id
+            for device_id in self._previous_devices
+            if config_entry.entry_id not in device_id
+        ]:
+            device_registry = dr.async_get(self.hass)
```

Comment:
Instead, set the minor_version of your config flow to 2 and implement `async_migrate_entry` in `__init__` to centralize the migration logic